drawee/src/main/java/com/facebook/drawee/view/DraweeTransform.java

```diff
@@ -0,0 +1,83 @@
+package com.facebook.drawee.view;
+
+import android.animation.Animator;
+import android.animation.AnimatorListenerAdapter;
+import android.animation.ValueAnimator;
+import android.annotation.TargetApi;
+import android.graphics.Rect;
+import android.os.Build;
+import android.transition.Transition;
+import android.transition.TransitionValues;
+import android.view.ViewGroup;
+
+import com.facebook.drawee.drawable.ScalingUtils;
+
+@TargetApi(Build.VERSION_CODES.KITKAT)
+/**
+ * This Transition animates changes of {@link GenericDraweeView} between two ScaleTypes
+ *
+ * In combination with ChangeBounds, DraweeTransform allows GenericDraweeViews
+ * that change size, shape, or {@link ScalingUtils.ScaleType} to animate contents
+ * smoothly.
+ */
+public class DraweeTransform extends Transition {
```

Comment:
For ease of usage, let's make a static factory method here that will create the necessary TransitionSet:

```
public static TransitionSet createTransitionSet() {
  TransitionSet transitionSet = new TransitionSet();
  transitionSet.addTransition(new ChangeBounds());
  transitionSet.addTransition(new DraweeTransform());
  return transitionSet;
}
```